src/Validation/Validation.php

```diff
@@ -323,6 +323,80 @@ public static function compareWith($check, $field, $context)
     }
 
     /**
+     * Validates if two fields match with each other.
+     *
+     * If both fields have exactly the same value this method will return true.
+     *
+     * You may set option `'fieldsExpected'` to `false` to pass the rule if either of
+     * the given fields miss in the context. Defaults to `true`.
+     *
+     * You may set option `'typeSafe'` to `false`, to omit type checking when matching. Defaults to `true`.
+     *
+     * @param string $fieldOne The first field to compare against the second field.
+     * @param string $fieldTwo The second field to compare against the first field.
+     * @param array $context The validation context.
+     * @param array $options Optional options for the check. Default to ['fieldsExpected' => true, 'typeSafe' => true].
+     * @return bool
+     */
+    public static function fieldsMatching($fieldOne, $fieldTwo, $context, array $options = [])
```

Comment:
This function should live in `Validator.php`. I see this file as a legacy thing that should be drained out of methods instead of adding more.

`Validator.php` is the place where you can guarantee the `$context` being passed correctly.